3.processing_features/scripts/0.merge_sc_cytotable.py

```diff
@@ -111,11 +111,35 @@
     print(f"Added single cell count as metadata to {pathlib.Path(dest_path).name}!")
 
 
-# ### Check if converted data looks correct
+# ## Update the files to remove NA rows added as artifacts of CytoTable
 
 # In[5]:
 
 
+for file_path in output_dir.iterdir():
+    if file_path.stem not in ["Plate_1", "Plate_2"]:
+        # Load the DataFrame from the Parquet file
+        df = pd.read_parquet(file_path)
+
+        # If any, drop rows where "Metadata_ImageNumber" is NaN (artifact of cytotable)
+        df = df.dropna(subset=["Metadata_ImageNumber"])
+
+        # Columns to move to the front
+        columns_to_move = ['Nuclei_Location_Center_X', 'Nuclei_Location_Center_Y', 'Cells_Location_Center_X', 'Cells_Location_Center_Y']
+
+        # Rearrange columns and add "Metadata" prefix in one line
+        df = (df[columns_to_move + [col for col in df.columns if col not in columns_to_move]]
+                    .rename(columns=lambda col: 'Metadata_' + col if col in columns_to_move else col))
+
+        # Save the processed DataFrame as Parquet in the same path
+        df.to_parquet(file_path, index=False)
+
+
+# ### Check if converted data looks correct
```

Comment:
you can also consider adding an `assert` statement here to confirm. Something like:

```python
assert df.Metadata_ImageNumber.isna().sum() == 0
```

Reply:
I'll look to implement this in a future PR so I will add as a issue for now.